### ISSUE TITLE
Discussion/Feature Request: Plotting of complex values

### ISSUE THREAD

**Issue by cweickhmann** [pyqtgraph/pyqtgraph#1269]
I've searched a fair bit in the docs and maybe I have overlooked this, but it seems there is no way to supply a pyqtgraph.PlotDataItem with complex numbers, right?
However, I am in RF engineering and we constantly have the issue that our measurement results are complex valued and we like to switch between magnitude, real-part, phase-plot or magnitude dB, for instance.
I would like to set up a modification to pyqtgraph where the underlying DataItem accepts complex values and the plot widget then has a toolbar or such where you can switch between the various representations of complex values.

My questions therefore are:
* Anything to keep in mind when doing this?
* Are others maybe currently thinking about something similar?
* Have I just not found this feature, yet?

I'm happy for any feedback!

**Comment by 2xB** [comment]
I am not really sure I understand the question, so to get it right: Your goal is to visualize multiple sets of real-valued 2D data, i.e. `(x, Re(x))` and `(x, Im(x))`, and adding Qt controls (i.e. a `QComboBox`) for data selection, correct? In that case I am unsure how modifying `pyqtgraph` would help, but you might be helped with an example on how to use a `QComboBox` together with `pyqtgraph`?

**Comment by cweickhmann** [comment]
No, actually, I would like to extend the interface such that a plot widget would accept numpy.complex values directly (of course, this could be done using two rows of reals, too).
Using a combobox or toolbar buttons or whatever, I would like to switch the visualisation between the modes mentioned above. In the extended widget this would be done with a slot so it could be bound to anything that fits the final UI.

The reason I'm bringing this up is that this is a recurring use case for RF engineering. And I feel it could well be part of the library.

**Comment by 2xB** [comment]
So it is not a question on how to realize such a GUI with `pyqtgraph`, but about building it in to the library. Okay!

This could be realized as a high-level widget similar to the `ImageView` widget, which combines multiple widgets (in our case it would combine at least a `QComboBox`, a `PlotWidget` and a `Label`).

An issue with providing a default visualization of a complex data set is that I would come to think of quite a few possible visualizations that would be useful in different applications, such as
 * One plot showing both `Im(x)` against time and `Re(x)` against time in two different lines
 * One plot showing only `Im(x)` against time or `Re(x)` against time in one line, with further interface components to select which of both to show
 * One scatter plot showing `Im(x)` on the `x` and `Re(x)` on the `y` axis, using color code for time
 * Two plots, one showing `|x|` against time and one showing phase against time

Now it does not seem trivial to me to decide which of these would be what people expect when `pg.plot()` receives complex numbers.

Since all of this is quite high-level - every single of the above mentioned plots can be done already with `pyqtgraph` -, maybe implementing those as custom widgets in an example file would be helpful, as others could just use the code provided in something like `pyqtgraph/examples/complex.py`, maintaining reusability, but also allowing for customization in own projects?

**Comment by cweickhmann** [comment]
Making it an example sounds like a good option. I'll work on it.
